File: SDKs/sportsdata.io/mlb-v3-stats/1.0/go/pkg/models/operations/boxscore.go
package operations

type BoxScoreFormatEnum string

const (
	BoxScoreFormatEnumXML  BoxScoreFormatEnum = "XML"
	BoxScoreFormatEnumJSON BoxScoreFormatEnum = "JSON"
)

type BoxScorePathParams struct {
	Format BoxScoreFormatEnum `pathParam:"style=simple,explode=false,name=format"`
	Gameid string             `pathParam:"style=simple,explode=false,name=gameid"`
}

type BoxScoreRequest struct {
	PathParams BoxScorePathParams
}

type BoxScoreResponse struct {
	BoxScore    *interface{}
	ContentType string
	StatusCode  int64
}

File: SDKs/sportsdata.io/nba-v3-stats/1.0/typescript/src/sdk/models/operations/boxscore.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum BoxScoreFormatEnum {
    Xml = "XML",
    Json = "JSON"
}


export class BoxScorePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=format" })
  format: BoxScoreFormatEnum;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=gameid" })
  gameid: string;
}


export class BoxScoreRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: BoxScorePathParams;
}


export class BoxScoreResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  boxScore?: any;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
